### PR TITLE
Refine channels management in Proxy.

### PR DESCRIPTION
**Pull request by longjiquan**
Fix a issue that dml channels will be changed in order under concurrency.
Signed-off-by: longjiquan <jiquan.long@zilliz.com>
issue: #16961

### CODE REVIEW

File: internal/proxy/channels_mgr.go [hunk 1]
```diff
@@ -45,60 +44,67 @@ type channelsMgr interface {
...
+	vchans []vChan
+	pchans []pChan
+}
+
+type streamInfos struct {
+	channelInfos channelInfos
+	stream       msgstream.MsgStream
+}
+
+func removeDuplicate(ss []string) []string {
```
Comment (czs007) [inline internal/proxy/channels_mgr.go:58]: remove this.

File: internal/proxy/channels_mgr.go [hunk 2]
```diff
@@ -111,323 +117,175 @@ const (
...
-
-	mgr.deleteVChansByVIDs(ids)
-	mgr.deleteStreamByVIDs(ids)
-	for _, pc := range channels {
-		metrics.ProxyMsgStreamObjectsForPChan.WithLabelValues(strconv.FormatInt(Params.ProxyCfg.GetNodeID(), 10), pc).Dec()
-	}
+	mgr.mu.Lock()
+	defer mgr.mu.Unlock()
+	// no need to check if collection exists.
+	delete(mgr.infos, collectionID)
```
Comment (czs007) [inline internal/proxy/channels_mgr.go:280]: call Close of MsgStream Object to release resource.

File: internal/proxy/channels_mgr.go [hunk 3]
```diff
@@ -111,323 +117,175 @@ const (
...
+// removeAllStream remove all message stream.
 func (mgr *singleTypeChannelsMgr) removeAllStream() error {
-	mgr.deleteAllChannels()
-	mgr.deleteAllStream()
-	mgr.deleteAllVChans()
-	mgr.deleteAllCollection()
-
+	mgr.mu.Lock()
+	defer mgr.mu.Unlock()
+	mgr.infos = make(map[UniqueID]streamInfos)
```
Comment (czs007) [inline internal/proxy/channels_mgr.go:288]: call Close of  All MsgStream Object to release resource.

File: internal/proxy/channels_mgr.go (hunk 1 above)
Comment (longjiquan) [reply internal/proxy/channels_mgr.go:58]: OK.

File: internal/proxy/channels_mgr.go (hunk 2 above)
Comment (longjiquan) [reply internal/proxy/channels_mgr.go:280]: We'll close `MsgStream` in it's `Finalizer`.

File: internal/proxy/channels_mgr.go (hunk 3 above)
Comment (longjiquan) [reply internal/proxy/channels_mgr.go:288]: We'll close MsgStream in it's Finalizer.

**Pull request merged**